src/mod_block_incoming.erl

```diff
@@ -0,0 +1,298 @@
+%%%-------------------------------------------------------------------
+%%% File    : mod_block_incoming.erl
+%%% Author  : Devin Sills <dsills@skillz.com>
+%%% Created : 31 Jan 2022 by Devin Sills <dsills@skillz.com>
+%%%
+%%%----------------------------------------------------------------------
+
+-module(mod_block_incoming).
+-author('dsills@skillz.com').
+
+-behaviour(gen_server).
+-behaviour(gen_mod).
+
+%% gen_mod callbacks.
+-export([start/2, stop/1, reload/3, depends/2, mod_options/1, mod_opt_type/1]).
+
+%% gen_server callbacks.
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2, terminate/2, code_change/3]).
+
+%% ejabberd_commands callbacks.
+-export([get_commands_spec/0, add_blocking_user/2, remove_blocking_user/2, get_blocking_users/1, reset_blocking_users/1, clear_blocking_users/1]).
+
+%% hook handlers
+-export([filter_packet/1, filter_offline_msg/1, filter_subscription/2]).
+
+-include("ejabberd_commands.hrl").
+-include("logger.hrl").
+-include_lib("xmpp/include/xmpp.hrl").
+
+-define(HOOK_PRIORITY, 25).
+
+-define(COMMAND_TIMEOUT, timer:seconds(30)).
```

Comment:
This seems like it might be a long timeout when its just reading the state

Reply:
and since the fallback is safe/quick (reading from the original config), it seems like we can be okay with failing quickly if it's somehow backed up

Reply:
I agree, I switched it to 2 seconds

Reply:
I agree, I switched it to 2 seconds